app/scripts/services/serviceInstances.js

```diff
@@ -3,64 +3,105 @@
-      // - automatically remove the bindings
-      // - tell the user they must manually unbind before continue
-      return $uibModal.open({
+      modalInstance =  $uibModal.open({
         animation: true,
-        templateUrl: 'views/modals/confirm.html',
+        templateUrl: 'views/modals/delete-resource.html',
         controller: 'ConfirmModalController',
         resolve: {
           modalConfig: function() {
             return modalScope;
           }
         }
-      })
-      .result.then(function() {
+      });
+
+      return modalInstance.result.then(function() {
         NotificationsService.hideNotification("deprovision-service-error");
 
-        return DataService.delete({
-            group: 'servicecatalog.k8s.io',
-            resource: 'serviceinstances'
-          },
-          apiObject.metadata.name,
-          { namespace: apiObject.metadata.namespace },
-          { propagationPolicy: null
-          }) // TODO - remove once this is resolved https://github.com/kubernetes-incubator/service-catalog/issues/942
-          .then(function() {
-            NotificationsService.addNotification({
-              type: "success",
-              message: "Successfully deleted provisioned service " + apiObject.metadata.name + "."
-            });
-          }, function(err) {
-            NotificationsService.addNotification({
-              id: "deprovision-service-error",
-              type: "error",
-              message: "An error occurred while deleting provisioned service " + apiObject.metadata.name + ".",
-              details: getErrorDetails(err)
-            });
+        var context = {namespace: apiObject.metadata.namespace};
+        var resource = {
+          group: 'servicecatalog.k8s.io',
+          resource: 'serviceinstances'
+        };
+
+        // TODO - remove once this is resolved https://github.com/kubernetes-incubator/service-catalog/issues/942
+        var opts = {
+          propagationPolicy: null
+        };
+
+        return DataService.delete(resource, apiObject.metadata.name, context, opts).then(function() {
+          NotificationsService.addNotification({
+            type: "success",
+            message: "Successfully deleted provisioned service " + apiObject.metadata.name + "."
+          });
+        }, function(err) {
+          NotificationsService.addNotification({
+            id: "deprovision-service-error",
+            type: "error",
+            message: "An error occurred while deleting provisioned service " + apiObject.metadata.name + ".",
+            details: getErrorDetails(err)
           });
+        });
       });
-    }
+    };
+
+    var showUnableToDeprovisionDialog = function() {
+      return $uibModal.open({
+        animation: true,
+        templateUrl: 'views/modals/warning.html',
+        controller: 'WarningModalController',
+        resolve: {
+          modalConfig: function() {
+            return {
+              title: "Unable to Deprovision",
```

Comment:
We call the action "Delete" in the UI, so I'd be consistent in the messages we use (in the dialog and notifications).